main.rb

```diff
@@ -0,0 +1,172 @@
+require './person'
+require './teacher'
+require './student'
+require './rental'
+require './book'
+require './classroom'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def execute
+    puts 'Welcome to School Library App!'
+
+    loop do
+      main_options
+      print 'Please Select an Option to Proceed: '
+      option = gets.chomp
+      break if option == '7'
+
+      option_output option
+    end
+
+    puts 'Thank you for using this app!'
+  end
+
+  def main_options
+    puts
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    puts '*******************'
+  end
+
+  def option_output(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    puts 'There is no book. Please Add a Book!' if @books.length.zero?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 1
+  end
+
+  def list_all_people
+    puts 'There is no person. Please Add a Person!' if @people.length.zero?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
```

Comment:
- [ ] Please ensure that the variables you initialized in the `Person` class are the same you use when you call the `super` as there are some errors occurring because of that. Here, the name does not show up when I entered the `2` command which is the `List all people` command. Instead, everything is showing up as an object for the age. Please let's fix this.

<img width="476" alt="Loom_xHMlG8tedF" src="https://user-images.githubusercontent.com/81468675/153454578-e33cbf2a-1db1-4e3e-bb8e-7324a9b9202d.png">

Reply:
Thank you. Implemented.